### PR TITLE
Support auto-reloading of TLS config in C++ implementation

### PR DESCRIPTION
**Pull request by vekterli**
@havardpe @bjorncs please review

By default reloads every 60 minutes. This also reloads all peer
authorization rules. Files referenced by the TLS config are reloaded
transitively.

If reloading fails a warning will be logged and the existing config
will continue to be in effect until the next reload time.

### CODE REVIEW

**Review by bjorncs — APPROVED** [review]
Consider including the command used for generating the test credentials (in case they have to be re-generated with different parameters).

File: vespalib/src/vespa/vespalib/net/tls/auto_reloading_tls_crypto_engine.cpp [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+      _config_file_path(std::move(config_file_path)),
+      _current_engine(tls_engine_from_config_file(_config_file_path)),
+      _reload_interval(reload_interval),
+      _reload_thread([this](){ run_reload_loop(); })
+{
+}
+
+AutoReloadingTlsCryptoEngine::~AutoReloadingTlsCryptoEngine() {
+    {
+        std::unique_lock lock(_mutex);
```
Comment (havardpe) [inline vespalib/src/vespa/vespalib/net/tls/auto_reloading_tls_crypto_engine.cpp]: consider using lock_guard

**Review by havardpe — APPROVED** [review]
👍

**Comment by vekterli** [comment]
Will address comments in a follow-up

**Pull request merged**